src/main/webapp/js/lib/utils.js

```diff
@@ -0,0 +1,77 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Creates spinner to signify loading and adds to the courseContainer
+ * @param {Element} courseContainer container for course list
+ */
+function attachNewSpinner(courseContainer) {
+  const spinner = document.createElement('i');
+  spinner.setAttribute('class', 'fas fa-sync fa-spin');
+  courseContainer.innerText = '';
+  courseContainer.appendChild(spinner);
+}
+
+/**
+ * Creates table from a 2D array
+ * @param {Object} tableData 2D array with separation of courses
+ * @param {Object} creditsData Array with credits for each semester
+ * @param {Element} courseContainer container for course list
+ */
+function createTable(tableData, creditsData, courseContainer) {
```

Comment:
The name seems more generic than it should be. It only creates one type of table. Are you sure this belongs in this util file? If so, you might want to rename the function name to be clearer too.